src/bb_excel/core.clj

```diff
@@ -132,68 +138,73 @@
-      (style-check u styles pcts)   (assoc-in u [:d] (num2pct (:d u)))
-      (style-check u styles dates)  (assoc-in u [:d] (num2date (:d u)))
-      (style-check u styles times)  (assoc-in u [:d] (num2time (:d u)))
-      :else u)))
+      ;; Possible cell-value types well explained here https://stackoverflow.com/a/18346273
+      (= cell-type "s")                 (assoc cell* :d (nth shared-strings (parse-long cell-value)))
+      (= cell-type "str")               cell*
+      (= cell-type "inlineStr")         cell*
+      (= cell-type "b")                 (assoc cell* :d (if (= "1" cell-value) true false))
+      (= cell-type "e")                 (assoc cell* :d (get error-codes cell-value))
+      (= cell-type "n")                 (assoc cell* :d (parse-long cell-value))
+      (style-check cell* styles pcts)   (assoc cell* :d (num2pct cell-value))
+      (style-check cell* styles dates)  (assoc cell* :d (num2date cell-value))
+      (style-check cell* styles times)  (assoc cell* :d (num2time cell-value))
+      :else cell*)))
+
+(defn- get-row-index
+  [row]
+  (parse-long (:y row)))
 
 (defn process-row
   "Process Excel row of data"
-  [dict styles coll]
-  (reduce #(merge % {:_r (read-string (:y %2)) (keyword (:x %2)) (:d %2)}) {}
-          (map (partial process-cell dict styles)
-               (map #(merge (first %) {:d (second %)}
-                            {:f (nth % 2)})
-                    (map (juxt :attrs
-                               (comp last :content last :content)
-                               (comp first :content first :content)) coll)))))
+  [shared-strings styles row]
+  (let [row* (->> row
+                  (map (fn [cell] (merge (:attrs cell)
+                                         {:d (-> cell :content last :content last)})))
+                  (map (partial process-cell shared-strings styles)))
+        row-index (get-row-index (first row*))]
+    (into {:_r row-index} (map #(-> [(keyword (:x %)) (:d %)]))
+          row*)))
 
 (defn- get-cell-text
-  "Extract "
-  [coll]
-  (apply str
-         (mapcat :content
-                 (filter #((:text-t tags) (:tag %))
-                         (xml-seq coll)))))
-
-(defn get-unique-strings
+  "Extract text from cell"
+  [cell]
+  (->> (xml-seq cell)
+       (filter #(contains? TEXT_T_TAGS (:tag %)))
+       (mapcat :content)
+       (str/join)))
+
+(defn get-shared-strings
   "Get dictionary of all unique strings in the Excel spreadsheet"
   [^ZipFile zipfile]
-  (if-let [wb (.getEntry zipfile (str "xl/sharedStrings.xml"))]
+  (if-let [wb (.getEntry zipfile "xl/sharedStrings.xml")]
     (let [ins (.getInputStream zipfile wb)
           x (parse-str (slurp ins))]
-      (->>
-       (filter #((:text-part tags) (:tag %)) (xml-seq x))
-       (map get-cell-text)
-       (zipmap (range))))
-    {}))
+      (into [] (comp (filter #(contains? TEXT_PART_TAGS (:tag %)))
+                     (map get-cell-text))
+            (xml-seq x)))
```

Comment:
Actually we don't need map with indexes. Vector make access by index in O(1) time with nth. Increase perfomance 20% on my sample.